source/common/common/thread.h

```diff
@@ -57,26 +58,40 @@ class MutexBasicLockable : public BasicLockable {
  */
 class CondVar {
 public:
-  // Note that it is not necessary to be holding an associated mutex to call
-  // notifyOne or notifyAll. See the discussion in
-  //     http://en.cppreference.com/w/cpp/thread/condition_variable_any/notify_one
-  // for more details.
+  enum WaitStatus {
+    Timeout,
+    NoTimeout,
```

Comment:
I think this would be clearer if it were named "Success"

Reply:
The problem is that this value may also represent a spurious wakeup.  I added a brief comment to that effect. You can only determine success if you know the bool condition for that in control of the application.